src/main/java/org/elasticsearch/common/settings/ImmutableSettings.java

```diff
@@ -1100,6 +1102,17 @@ public boolean shouldIgnoreMissing(String placeholderName) {
         }
 
         /**
+         * Runs through all settings in this builder and prompts the user to provide a value for each <code>__prompt__</code>
+         * value that is found in the settings.
+         * @param terminal The terminal to user to for input/output
+         * @return The builder
+         */
+        public Builder replacePromptPlaceholders(Terminal terminal) {
```

Comment:
Yeah it does make sense. The static map was more for issues that occurred when creating a `Node` within `Bootstrap` and having system properties with `__prompt__`. We can workaround that by not re-reading the system properties when building the node.